network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/NodeBreakerViewImpl.java

```diff
@@ -100,8 +100,11 @@ public int getBusbarSectionCount() {
     @Override
     public BusbarSection getBusbarSection(String id) {
         checkBusBreakerTopology();
-        return index.getBusbarSection(id)
-                .orElse(null);
+        BusbarSection bbs = index.getBusbarSection(id).orElse(null);
+        if (bbs != null && bbs.getTerminal().getVoltageLevel().getId() != this.voltageLevel.getId()) {
```

Comment:
You must use equals to compare the strings

Reply:
fixed